proto/internal/core/v1/core.proto

```diff
@@ -7,23 +7,11 @@ import "google/protobuf/any.proto";
 import "validate/validate.proto";
 
 message RelationTuple {
-  // Each tupleset specifies keys of a set of relation tuples. The set can
-  // include a single tuple key, or all tuples with a given object ID or
-  // userset in a namespace, optionally constrained by a relation name.
-  //
-  // examples:
-  // doc:readme#viewer@group:eng#member (fully specified)
-  // doc:*#*#group:eng#member (all tuples that this userset relates to)
-  // doc:12345#*#* (all tuples with a direct relationship to a document)
-  // doc:12345#writer#* (all tuples with direct write relationship with the
-  // document) doc:#writer#group:eng#member (all tuples that eng group has write
-  // relationship)
-
-  /** object_and_relation is the resource for the tuple */
-  ObjectAndRelation object_and_relation = 1 [ (validate.rules).message.required = true ];
-
-  /** user is the subject for the tuple */
-  User user = 2 [ (validate.rules).message.required = true ];
+  /** resource_and_relation is the resource for the tuple */
+  ObjectAndRelation resource_and_relation = 1 [ (validate.rules).message.required = true ];
+
+  /** subject is the subject for the tuple */
+  ObjectAndRelation subject = 2 [ (validate.rules).message.required = true ];
```

Comment:
This would seem like a deliberate API breakage, and presumably we are OK with that? Would this cause downtime to clients as the server with the new proto definition is rolled out?

Reply:
Referenced below